.github/renovate.json

```diff
@@ -24,10 +51,47 @@
       ],
       "matchStringsStrategy": "any",
       "matchStrings": [
-        ".*?\"downloadURL\":\\s*\"(?<registryUrl>[^/]+)/(?<packageName>[^:]+)[\\s\\S]*?\"multiArchVersions\": \\[\\s*\"(?<currentValue>[^\\\"]+)\".*"
+        "\"renovateTag\":\\s*\"registry=(?<registryUrl>[^,]+),name=(?<packageName>[^\"]+)\",\\s*\"latestVersion\":\\s*\"(?<currentValue>[^\"]+)\"(?:[^}]*\"previousLatestVersion\":\\s*\"(?<depType>[^\"]+)\")?"
       ],
       "datasourceTemplate": "docker",
```

Comment:
should we avoid using the `template` ? 

What is the delimiter between all of the custom Regex to ensure they don't pickup the same lines ?

Reply:
`should we avoid using the template ?` => do you mean we should extract datasource from the comments/RenovateTag? or You mean something else?

Reply:
The types of matchStrings we have now are as follows: (converted to human readable format)
1. "renovateTag": "registry=<RegistryURL>, name=<ContainerImage Name>"
2. "renovateTag": "name=moby-containerd, os=<OS>, release=<Release>"
     - for this type, we have os=ubuntu, release = 18.04|20.04|22.04|24.04
We need to always make sure each custom Regex only matches 1 type of renovateTag so that it won't pick up the same lines.

Reply:
should we avoid using the template ? => do you mean we should extract datasource from the comments/RenovateTag? or You mean something else?

- correct, unless it will always be docker

Reply:
The types of matchStrings we have now are as follows:

-- can we document this somerwhere ? I know it's in the code, but maybe finding a place to verbalize it will help others looking at updating that file.

Reply:
yeah the `containerImages` are always "docker". If it moves to somewhere else in the future, we can update the revonate.json and components.json.
And for `packages`, the custom managers are defined differently. So we still don't need to extract dataSource from RenovateTag. I will keep it as is until we see a demand to abstract the custom managers.
I will document all these.